examples/flutter_notifications/lib/main.dart

```diff
@@ -0,0 +1,61 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_notifications/notification/bloc/notification_bloc.dart';
+import 'package:flutter_notifications/notification/view/screen/notification_page.dart';
+import 'package:notification_repository/notification_repository.dart';
+
+import 'home.dart';
+
+void main() async {
+  WidgetsFlutterBinding.ensureInitialized();
+  NotificationRepositoryFCM notificationRepositoryFCM =
+      NotificationRepositoryFCM();
+  await notificationRepositoryFCM.initialize();
+  runApp(MyApp(
+    notificationRepositoryFCM: notificationRepositoryFCM,
+  ));
+}
+
+class MyApp extends StatefulWidget {
+  MyApp({Key? key, required this.notificationRepositoryFCM}) : super(key: key);
+
+  final NotificationRepositoryFCM notificationRepositoryFCM;
+
+  @override
+  _MyAppState createState() => _MyAppState();
+}
+
+class _MyAppState extends State<MyApp> {
+  final _navigatorKey = GlobalKey<NavigatorState>();
+
+  NavigatorState? get _navigator => _navigatorKey.currentState;
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocProvider<NotificationBloc>(
```

Comment:
Can we move the dependency injection to the App widget and consume the bloc you create at an AppView widget? That way, we avoid using the builder property from `MaterialApp`.

You can check this DI pattern in [this](https://github.com/felangel/bloc/blob/master/examples/flutter_weather/lib/weather/view/weather_page.dart) example with `WeatherPage` and `WeatherView`.